javatests/google/registry/rde/XjcToDomainResourceConverterTest.java

```diff
@@ -0,0 +1,326 @@
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests for {@link XjcToDomainResourceConverter} */
+@RunWith(JUnit4.class)
+public class XjcToDomainResourceConverterTest {
+
+  private static final ByteSource DOMAIN_XML = RdeTestData.get("domain_fragment.xml");
+  private static final ByteSource DOMAIN_XML_ADD_PERIOD =
+      RdeTestData.get("domain_fragment_addPeriod.xml");
+  private static final ByteSource DOMAIN_XML_AUTO_RENEW_PERIOD =
+      RdeTestData.get("domain_fragment_autoRenewPeriod.xml");
+  private static final ByteSource DOMAIN_XML_REDEMPTION_PERIOD =
+      RdeTestData.get("domain_fragment_redemptionPeriod.xml");
+  private static final ByteSource DOMAIN_XML_RENEW_PERIOD =
+      RdeTestData.get("domain_fragment_renewPeriod.xml");
+  private static final ByteSource DOMAIN_XML_PENDING_DELETE_PERIOD =
+      RdeTestData.get("domain_fragment_pendingDeletePeriod.xml");
+  private static final ByteSource DOMAIN_XML_PENDING_RESTORE_PERIOD =
+      RdeTestData.get("domain_fragment_pendingRestorePeriod.xml");
+  private static final ByteSource DOMAIN_XML_TRANSFER_PERIOD =
+      RdeTestData.get("domain_fragment_transferPeriod.xml");
+  private static final ByteSource DOMAIN_XML_UPDATE_REGISTRAR = RdeTestData.get(
+      "domain_fragment_up_rr.xml");
+  private static final ByteSource DOMAIN_XML_WITH_HOST_OBJS = RdeTestData.get(
+      "domain_fragment_host_objs.xml");
+  private static final ByteSource DOMAIN_XML_WITH_HOST_ATTRS = RdeTestData.get(
+      "domain_fragment_host_attrs.xml");
+  private static final ByteSource DOMAIN_XML_REGISTRANT_MISSING = RdeTestData.get(
+      "domain_fragment_registrant_missing.xml");
+  private static final ByteSource DOMAIN_XML_SECDNS = RdeTestData.get(
+      "domain_fragment_secdns.xml");
+
+  //List of packages to initialize JAXBContext
+  private static final String JAXB_CONTEXT_PACKAGES = Joiner.on(":")
+      .join(Arrays.asList(
+          "google.registry.xjc.contact",
+          "google.registry.xjc.domain",
+          "google.registry.xjc.host",
+          "google.registry.xjc.mark",
+          "google.registry.xjc.rde",
+          "google.registry.xjc.rdecontact",
+          "google.registry.xjc.rdedomain",
+          "google.registry.xjc.rdeeppparams",
+          "google.registry.xjc.rdeheader",
+          "google.registry.xjc.rdeidn",
+          "google.registry.xjc.rdenndn",
+          "google.registry.xjc.rderegistrar",
+          "google.registry.xjc.smd"));
+
+  @Rule
+  public final AppEngineRule appEngine = AppEngineRule.builder()
+      .withDatastore()
+      .build();
+
+  @Rule
+  public final ExceptionRule thrown = new ExceptionRule();
+
+  private Unmarshaller unmarshaller;
+
+  @Before
+  public void before() throws Exception {
+    createTld("example");
+    unmarshaller = JAXBContext.newInstance(JAXB_CONTEXT_PACKAGES)
+        .createUnmarshaller();
+  }
+
+  @Test
+  public void testConvertDomainResource() throws Exception {
```

Comment:
I don't see any assertions that dependent entities (e.g. history entries, poll messages, billing events) are created correctly.  That definitely seems worth testing because it's hugely important.